### PR TITLE
refactor: move public specific components into public site folder

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue Overview

This PR addresses #3238

- [x] This change addresses only certain aspects of the issue

## Description

This PR moves some components out of UIC and into the public site that are specific to only the public site. I have tabled `ApplicationTimeline` for now because it will be a bigger effort - the `ApplicationTimeline` component has a strange approach where it pulls in class names from the `ProgressNav` and overwrites them but doesn't do so with V2 styling (followup [issue](https://github.com/bloom-housing/bloom/issues/3259)).

It also reorganizes the public site directory - because we are adding more individual components, leaving it as-is was just a little confusing. The update is based on best practices from perusing articles. The directory changes are as follows:

Screenshot 2023-01-11 at 1 57 36 PM

* All site content now lives under the `/src` directory.
* There is a new `components` folder organized by each section of the application. This is where the shuffled components ended up. I would encourage us to move some of our contained UI pieces into this folder to encourage modularization, and it would make it a lot easier to test them (lookin at you `ListingView.tsx`).
* One of the components brought over had its own scss file. For file-level CSS, NextJs by default supports CSS modules. You can see this pattern with `StatusItem`. I would want to remove Tailwind and continue to lean on CSS modules to keep ourselves tidy with our own design system.

## How Can This Be Tested/Reviewed?

The only code change was in the `StatusItem` component, since it was moved to use CSS modules (this component renders on the my applications page within accounts and is a summary of a submitted application). I would ensure there are not visual regressions by submitting an application and checking out the component.

Otherwise, take a look at the directory structure and open to feedback about how to make things more organized / readable!

There are six test files from the components brought over in a new `__tests__` folder that should also be running successfully in a new job below (`unit-test-public`).

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Review by ludtkemorgan — APPROVED** [review]
So far this looks good. I haven't found any issues in my testing. Two questions though:

- I'm guessing this was done intentionally, but I noticed with this change we no longer have storybook files for the components moved over from ui-components. I'm ok with us no longer having them, but just wanted to make sure that was intentional
- I noticed the coverage.ts file which I haven't noticed before. Is that just for when running `test:coverage` in the public directory? If so, how often is that run?

File: sites/public/__tests__/components/listing/GetApplication.test.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+          street2: "Street 2",
+          street: "Pick Up Address Street",
+          zipCode: "90210",
+        }}
+        preview={false}
+      />
+    )
+    expect(getByText("How to Apply")).toBeTruthy()
+    expect(getByText("Apply Online")).toBeTruthy()
+    expect(getByText("Apply Online").closest("a")?.getAttribute("href")).toBe("online-app-url")
```
Comment (ludtkemorgan) [inline sites/public/__tests__/components/listing/GetApplication.test.tsx:33]: I know these are existing tests and what is here works so no need to change. But to do a more modern approach of tackling this you could do
`expect(getByRole("link", { name: "Apply Online" })).toHaveAttribute("href", "online-app-url")`.

Also all of the `toBeTruthy()` can be `toBeInTheDocument()`

File: sites/public/__tests__/components/listing/GetApplication.test.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+          city: "City",
+          state: "State",
+          street2: "Street 2",
+          street: "Pick Up Address Street",
+          zipCode: "90210",
+        }}
+        preview={true}
+      />
+    )
+    expect(getByText("Apply Online").closest("button")?.disabled).toBe(true)
```
Comment (ludtkemorgan) [inline sites/public/__tests__/components/listing/GetApplication.test.tsx:85]: This one as well could be `expect(getByRole("button", { name: "Apply Online" })).toBeDisabled()`. But no need to change for this issue

**Review by ColinBuyck — APPROVED** [review]
This LGTM 🪅  once Morgan's comments are resolved

**Comment by emilyjablonski** [comment]
@ludtkemorgan Thanks for the review!

Removal of stories was intentional - bc these aren't in UIC anymore, we can't have stories for what was moved to public, unless we were to decide we want Storybook in public as well - but since the motivating reason to move them is that they aren't super component-y I think we're good to remove

And then I believe that file is used by cypress when creating code coverage reports, so I don't think it's when `test:coverage` is run.

**Pull request merged**